include/nlohmann/json.hpp

```diff
@@ -1212,15 +1212,13 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     /// @brief move constructor
     /// @sa https://json.nlohmann.me/api/basic_json/basic_json/
     basic_json(basic_json&& other) noexcept
-        : json_base_class_t(std::forward<json_base_class_t>(other)),
-          m_data(std::move(other.m_data))
+    // check that passed value is valid (has to be done before forwarding)
+        : json_base_class_t((other.assert_invariant(false), std::forward<json_base_class_t>(other))),
```

Comment:
Can we just do `assert_invariant(false);` in the body (assert it after moving) to avoid this mess?

Reply:
Another possibility for avoiding all these issues is to do `std::swap()` on `m_data` and `other.m_data` in the body.

Reply:
I am not at all happy with this. I also thought about swapping...

Reply:
Sorry for asking a question for something that is probably obvious, but which constructor of `json_base_class_t` is this call valid for? I suppose it is one of the `basic_json` constructors being used here?

FWIW, it should be completely fine to check the invariants after the initializer list has been evaluated. Is there a possibility to consider rewriting the `data` move constructor so these assignments don't need to be done at this level?